components/prism-javascript.js

```diff
@@ -14,9 +14,25 @@ Prism.languages.insertBefore('javascript', 'keyword', {
 	},
 	// This must be declared before keyword because we use "function" inside the look-forward
 	'function-variable': {
-		pattern: /[_$a-z\xA0-\uFFFF][$\w\xA0-\uFFFF]*(?=\s*=\s*(?:function\b|(?:\([^()]*\)|[_$a-z\xA0-\uFFFF][$\w\xA0-\uFFFF]*)\s*=>))/i,
+		pattern: /[_$a-z\xA0-\uFFFF][$\w\xA0-\uFFFF]*(?=\s*=\s*(?:async\s*)?(?:function\b|(?:\([^()]*\)|[_$a-z\xA0-\uFFFF][$\w\xA0-\uFFFF]*)\s*=>))/i,
```

Comment:
Shouldn't this be `async\s+` ?

Reply:
If we use `async\s+`, we couldn't match the variable `foo` in `foo = async(a, b) =>`, where the `async` here is a reserve word instead of the function name.

I found a better solution `async(?:(?:\s*(?=\())|\s+)`.